management/persistence.md

```diff
@@ -0,0 +1,330 @@
+
+### Log rewriting
+
+The AOF gets bigger and bigger as write operations are
+performed.  For example, if you are incrementing a counter 100 times,
+you'll end up with a single key in your dataset containing the final
+value, but 100 entries in your AOF. 99 of those entries are not needed
+to rebuild the current state.
+
+The rewrite is completely safe.
+While FalkorDB continues appending to the old file,
+a completely new one is produced with the minimal set of operations needed to create the current data set,
+and once this second file is ready FalkorDB switches the two and starts appending to the new one.
+
+FalkorDB supports an interesting feature: it is able to rebuild the AOF in the background without interrupting service to clients.
+Whenever you issue a `BGREWRITEAOF`, FalkorDB will write the shortest sequence of commands needed to rebuild the current dataset in memory.
+FalkorDB will automatically trigger log rewriting automatically (see the example configuration file for more information).
+
+When an AOF rewrite is scheduled, the FalkorDB parent process opens a new incremental AOF file to continue writing.
+The child process executes the rewrite logic and generates a new base AOF.
+FalkorDB will use a temporary manifest file to track the newly generated base file and incremental file.
+When they are ready, FalkorDB will perform an atomic replacement operation to make this temporary manifest file take effect.
+In order to avoid the problem of creating many incremental files in case of repeated failures and retries of an AOF rewrite,
+FalkorDB introduces an AOF rewrite limiting mechanism to ensure that failed AOF rewrites are retried at a slower and slower rate.
+
+### How durable is the append only file?
+
+You can configure how many times FalkorDB will
+[`fsync`](https://linux.die.net/man/2/fsync) data on disk. There are
+three options:
+
+* `appendfsync always`: `fsync` every time new commands are appended to the AOF. Very very slow, very safe. Note that the commands are appended to the AOF after a batch of commands from multiple clients or a pipeline are executed, so it means a single write and a single fsync (before sending the replies).
+* `appendfsync everysec`: `fsync` every second. Fast enough and you may lose 1 second of data if there is a disaster.
+* `appendfsync no`: Never `fsync`, just put your data in the hands of the Operating System. The faster and less safe method. Normally Linux will flush data every 30 seconds with this configuration, but it's up to the kernel's exact tuning.
+
+The suggested (and default) policy is to `fsync` every second. It is
+both fast and relatively safe. The `always` policy is very slow in
+practice, but it supports group commit, so if there are multiple parallel
+writes FalkorDB will try to perform a single `fsync` operation.
+
+### What should I do if my AOF gets truncated?
```

Comment:
Everything paragraph related to persistency corruption should be moved to a different page:  e.g. `persistency troubleshooting`